taichi/program/ndarray.cpp

```diff
@@ -31,6 +36,10 @@ intptr_t Ndarray::get_data_ptr_as_int() const {
   return reinterpret_cast<intptr_t>(data_ptr_);
 }
 
+intptr_t Ndarray::get_device_allocation_ptr_as_int() const {
+  return reinterpret_cast<intptr_t>(device_allocation_ptr_);
```

Comment:
super nit here:

1. If we don't store this as an additional field, we can just `reinterpret_cast<intptr_t>(&device_allocation_ptr_)`
2. If we do, why not just store it as an `intptr_t`...

Reply:
Right... I was dumb....thanks for catching this!